### PR TITLE
Migrate off datamodel

### PR DESCRIPTION
**Pull request by brinxmat**
In this PR:

- Remove dependency on nva-datamodel-java
- Update tests so that everything is tested from handler
- Add tests where necessary
- Some small updates to code style
- Identify problem areas
  - No not-found checks for files in S3 (not sure how to handle so open to discussion)
  - No handling of embargoes (should only be downloadable for owner) [NP-3395](https://unit.atlassian.net/browse/NP-3395)
  - No handling of administrative files (should only be downloadable for owner) [NP-3396](https://unit.atlassian.net/browse/NP-3396)
  - Further, curator scope is not managed, but this is another story

### CODE REVIEW

File: create-presigned-download-url/src/test/java/no/unit/nva/download/publication/file/CreatePresignedDownloadUrlHandlerTest.java [hunk 1]
```diff
@@ -1,397 +1,532 @@
...
-            .thenThrow(new NotFoundException(ERROR_PUBLICATION_NOT_FOUND_FOR_IDENTIFIER + IDENTIFIER_VALUE));
-        createPresignedDownloadUrlHandler.handleRequest(inputStream(IDENTIFIER, IDENTIFIER_FILE_VALUE), output,
-                context);
-
-        GatewayResponse<CreatePresignedDownloadUrlResponse> gatewayResponse = GatewayResponse.fromOutputStream(output);
-        assertEquals(HttpURLConnection.HTTP_NOT_FOUND, gatewayResponse.getStatusCode());
-        assertTrue(gatewayResponse.getHeaders().containsKey(CONTENT_TYPE));
-        assertTrue(gatewayResponse.getHeaders().containsKey(ACCESS_CONTROL_ALLOW_ORIGIN));
+    void shouldReturnNotFoundResponseOnUnknownFileIdentifier() {
+    }
```
Comment (brinxmat) [inline create-presigned-download-url/src/test/java/no/unit/nva/download/publication/file/CreatePresignedDownloadUrlHandlerTest.java:217]: A bit weird that this isn't in place, but there is no check that the file actually exists…

File: build.gradle [hunk 2]
```diff
@@ -33,15 +33,16 @@ subprojects {
     def awsSdkVersion = '1.12.99'
     def zalandoVersion = '0.26.0'
     def mockitoVersion = '4.0.0'
-    def nvaCommonsVersion = '1.16.1'
+    def nvaCommonsVersion = '1.16.8'
```
Comment (axthosarouris) [inline build.gradle:36]: 1.16.9

File: create-presigned-download-url/src/main/java/no/unit/nva/download/publication/file/CreatePresignedDownloadUrlHandler.java [hunk 3]
```diff
@@ -52,36 +52,41 @@ public CreatePresignedDownloadUrlHandler() {
...
-            String identifier = RequestUtil.getIdentifier(requestInfo);
-            String userId = RequestUtil.getUserIdOptional(requestInfo).orElse(null);
-            authorize(identifier, userId, publication);
+            authorize(identifier, user, publication);
         }
     }
 
     @Override
-    protected Integer getSuccessStatusCode(Void input, CreatePresignedDownloadUrlResponse output) {
+    protected Integer getSuccessStatusCode(Void input, PresignedUriResponse output) {
```
Comment (axthosarouris) [inline create-presigned-download-url/src/main/java/no/unit/nva/download/publication/file/CreatePresignedDownloadUrlHandler.java:84]: HttpURLConnection.HTTP_OK. I do not think we need to create a dependency to the apache library for this.

File: create-presigned-download-url/src/test/java/no/unit/nva/download/publication/file/publication/PublicationBuilder.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+     * @return JSON string representing an API response.
+     */
+    public String build() {
+        return getTemplateAsString()
+                .replace("__STATUS__", status.name())
+                .replace("__IDENTIFIER__", identifier)
+                .replace("__OWNER__", owner)
+                .replace("__ORGANIZATION__", organization)
+                .replace("__FILE_IDENTIFIER__", fileIdentifier)
+                .replace("\"__MIME_TYPE__\"", mimeType);
```
Comment (axthosarouris) [inline create-presigned-download-url/src/test/java/no/unit/nva/download/publication/file/publication/PublicationBuilder.java:67]: I think if you remove the `\"` from here, you will be able to remove the `\"` from line 52 as well.

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: create-presigned-download-url/src/test/java/no/unit/nva/download/publication/file/publication/PublicationBuilder.java (hunk 4 above)
Comment (brinxmat) [reply create-presigned-download-url/src/test/java/no/unit/nva/download/publication/file/publication/PublicationBuilder.java:67]: Unfortunately, this is a tedious fix for codacy complaining that the JSON is not valid JSON in the template (I want the template to be JSON for auto-formatting purposes)

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**